core/src/main/java/org/apache/iceberg/ReachableFileUtil.java

```diff
@@ -137,11 +139,21 @@ public static List<String> manifestListLocations(Table table, Set<Long> snapshot
    * @return the location of statistics files
    */
   public static List<String> statisticsFilesLocations(Table table) {
-    List<String> statisticsFilesLocations = Lists.newArrayList();
-    for (StatisticsFile statisticsFile : table.statisticsFiles()) {
-      statisticsFilesLocations.add(statisticsFile.path());
-    }
+    return statisticsFilesLocations(table, statisticsFile -> true);
+  }
 
-    return statisticsFilesLocations;
+  /**
+   * Returns locations of statistics files for a table matching the given predicate .
+   *
+   * @param table table for which statistics files needs to be listed
+   * @param predicate predicate for filtering the statistics files
+   * @return the location of statistics files
+   */
+  public static List<String> statisticsFilesLocations(
+      Table table, Predicate<StatisticsFile> predicate) {
```

Comment:
nit: is a newline required here after shorten the variable name?

Reply:
yes required. I tried keeping in the same line but the spotless make it into newline as line break is 100 char I guess.